docs/includes/metrics/metrics-shared.adoc

```diff
@@ -0,0 +1,171 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+// tag::overview[]
+
+ifndef::rootdir[:rootdir: {docdir}/../..]
+ifndef::flavor-lc[:flavor-lc: se]
+:description: Helidon metrics
+:keywords: helidon, metrics
+:writing-code-content: code which explicitly invokes the metrics API to register metrics, retrieve previously-registered metrics, and update metric values.
+
+* a unified way for
+ifdef::mp-flavor[MicroProfile]
+ifdef::se-flavor[Helidon]
+servers to export monitoring data--telemetry--to management agents, and
+* a unified Java API which all application programmers can use to register and update metrics to expose telemetry data from their services.
+ifdef::mp-flavor[]
+* support for metrics-related annotations.
+
+endif::[]
+Learn more about the https://github.com/eclipse/microprofile-metrics/releases/tag/{version-lib-microprofile-metrics-api}[MicroProfile Metrics specification].
+
+// end::overview[]
+
+// tag::usage-body[]
+=== Instrumenting Your Service
+
+You add metrics to your service
+ifdef::se-flavor[]
+by writing {writing-code-content}
+endif::[]
+ifdef::mp-flavor[]
+in these ways:
+
+* Annotate bean methods--typically your REST resource endpoint methods (the Java code that receives incoming REST requests); Helidon automatically registers these metrics and updates them when the annotated methods are invoked via CDI.
+* Write {writing-code-content}
+* Configure some simple `REST.request` metrics which Helidon automatically registers and updates for all REST resource endpoints.
+endif::[]
+
+Later sections of this document describe how to do
+ifdef::mp-flavor[each of these.]
+ifdef::se-flavor[this.]
+
+=== Categorizing Types of Metrics
+Helidon distinguishes among three general _types_, or scopes, of metrics, as described in the MP metrics specification.
+
+.Types (scopes) of metrics
+[%autowidth]
+|====
+| Type/scope | Typical Usage
+
+| base | Mandated by the MP metrics specification, such as OS or Java runtime measurements (available heap, disk space, etc.).
+| vendor | Implemented by vendors, including the `REST.request` metrics and other key performance indicator measurements (described in later sections).
+| application | Declared via annotations or programmatically registered by your service code.
+|====
+
+When you add metrics annotations to your service code, Helidon registers the resulting metrics as type `application`.
+
+=== Metric Registries
+A _metric registry_ collects registered metrics of a given type. Helidon supports three registries, one for each of the three metrics types.
+
+When you add code to your service to create a metric programmatically, the code first locates the appropriate registry and then the registers the metric with that registry.
+
+=== Retrieving Metrics Reports from your Service
```

Comment:
I've seen these level titles using sentence case in other docs

Reply:
I think we've been inconsistent on this, then. Many `.adoc` files have `===` and `====` headers in title case.

For now I will leave these in title case.